### ISSUE TITLE
No api function to replace all text on a specific line.

### ISSUE THREAD

**Issue by ACPixel** [adobe/brackets#12596]
like the title said, i did a ton of digging and could not find a function to replace a full line of text, or get all text on a specific line. im trying to write a neat little extension, but i need to be able to get all the text on a line whenever its changed, (basically on change event that outputs what line was changed and all thext on that line) and i need to be able to replace all text on a certain line. excuse me if this is the wrong place to post this, or if im missing something big. but i couldent seem to find anything, or anywhere else to post.

**Comment by petetnt** [comment]
Hi @ACPixel,

see the [documentation on `CodeMirror`](https://codemirror.net/doc/manual.html#commands) which provides the editor Brackets uses for replacing complete lines. Specifically:

> Whenever points in the document are represented, the API uses objects with line and ch properties. Both are zero-based. CodeMirror makes sure to 'clip' any positions passed by client code so that they fit inside the document, so you shouldn't worry too much about sanitizing your coordinates. **If you give ch a value of null, or don't specify it, it will be replaced with the length of the specified line.**

and 

> `doc.replaceRange(replacement: string, from: {line, ch}, to: {line, ch}, ?origin: string)`
Replace the part of the document between from and to with the given string. from and to must be `{line, ch}` objects. to can be left off to simply insert the string at position from. When origin is given, it will be passed on to "change" events, and its first letter will be used to determine whether this change can be merged with previous history events, in the way described for selection origins.
```

So basically the workflow would be:

```
1 . Get the active editor
2. Get the document on the editor
3. Find the line you want to replace (for example 10)
4. use `doc.replaceRange("Foo bar", from: {line: 10, ch: 0}, to: {line: 10})`
5. Selected line 10 should now say `Foo bar`.
```

**Comment by petetnt** [comment]
Closing as answered, feel free to reopen if the solution doesn't work for you.
